src/data/feature_index.js

```diff
@@ -41,6 +41,14 @@ class FeatureIndex {
             this.rawTileData = rawTileData;
             this.bucketLayerIDs = serialized.bucketLayerIDs;
             this.paintPropertyStatistics = serialized.paintPropertyStatistics;
+
+            // make a list of layer ids known to this FeatureIndex
+            this.knownLayers = {};
+            for (const index in this.bucketLayerIDs) {
+                for (const id of this.bucketLayerIDs[index]) {
+                    this.knownLayers[id] = true;
```

Comment:
I worry that the word "known" might make less sense once the context of this PR has faded. How 'bout just `layerIDs`? 

 - `bucketLayerIDs` -> layer ids grouped by bucket
 - `layerIDs` -> plain old layer ids.